lib/lndclient/LndClient.ts

```diff
@@ -382,19 +395,72 @@ class LndClient extends SwapClient {
   /**
    * Connects to another lnd node.
    */
-  public connectPeer = (pubkey: string, host: string, port: number): Promise<lndrpc.ConnectPeerResponse> => {
+  public connectPeer = (pubkey: string, address: string): Promise<lndrpc.ConnectPeerResponse> => {
     const request = new lndrpc.ConnectPeerRequest();
-    const address = new lndrpc.LightningAddress();
-    address.setHost(`${host}:${port}`);
-    address.setPubkey(pubkey);
-    request.setAddr(address);
+    const lightningAddress = new lndrpc.LightningAddress();
+    lightningAddress.setHost(address);
+    lightningAddress.setPubkey(pubkey);
+    request.setAddr(lightningAddress);
     return this.unaryCall<lndrpc.ConnectPeerRequest, lndrpc.ConnectPeerResponse>('connectPeer', request);
   }
 
+  /**
+   * Opens a channel given peerPubKey and amount.
+   */
+  public openChannel = async (
+    { peerIdentifier: peerPubKey, units, lndUris }:
+    { peerIdentifier: string, units: number, lndUris: string[] },
+  ): Promise<void> => {
+    const connectionEstablished = await this.connectPeerAddreses(lndUris);
+    if (connectionEstablished) {
+      await this.openChannelSync(peerPubKey, units);
+    } else {
+      throw new Error('connectPeerAddreses failed');
+    }
+  }
+
+  /**
+   * Tries to connect to a given list of peer's node addresses
+   * in a sequential order.
+   * Returns true when successful, otherwise false.
+   */
+  private connectPeerAddreses = async (
+    peerListeningUris: string[],
+  ): Promise<boolean> => {
+    const splitListeningUris = peerListeningUris
+      .map((uri) => {
+        const splitUri = uri.split('@');
+        return {
+          peerPubKey: splitUri[0],
+          address: splitUri[1],
+        };
+      });
+    const CONNECT_TIMEOUT = 4000;
+    for (const uri of splitListeningUris) {
+      const { peerPubKey, address } = uri;
+      let timeout;
+      try {
+        timeout = setTimeout(() => {
```

Comment:
I was thinking this might be a good spot to use `Promise.race` - but after looking at this a bit more the grpc calls actually take an options parameter with a `deadline` field to specify the max time for the call to complete. Doesn't have to be in this PR, but it'd be nice to use that (and maybe reuse it elsewhere down the road). I can create a quick followup PR perhaps.

Reply:
Hmm... perhaps. The benefit of having the timeout here would be a more clear error message.

I initially tried it with `Promise.race`, but I ran into some issues (can't remember the specifics).